clang/lib/Basic/Targets/SPIR.h

```diff
@@ -273,6 +273,29 @@ class LLVM_LIBRARY_VISIBILITY MicrosoftX86_64_SPIR64TargetInfo
     Builder.defineMacro("_M_AMD64", "100");
   }
 };
+
+// x86-64 SPIR64 Linux target
+class LLVM_LIBRARY_VISIBILITY LinuxX86_64_SPIR64TargetInfo
+    : public LinuxTargetInfo<SPIR64SYCLDeviceTargetInfo> {
+public:
+  LinuxX86_64_SPIR64TargetInfo(const llvm::Triple &Triple,
+                               const TargetOptions &Opts)
+      : LinuxTargetInfo<SPIR64SYCLDeviceTargetInfo>(Triple, Opts) {
+    LongDoubleWidth = LongDoubleAlign = 128;
+  }
+};
+
+// x86-32 SPIR Linux target
+class LLVM_LIBRARY_VISIBILITY LinuxX86_32_SPIRTargetInfo
+    : public LinuxTargetInfo<SPIR32SYCLDeviceTargetInfo> {
+public:
+  LinuxX86_32_SPIRTargetInfo(const llvm::Triple &Triple,
+                             const TargetOptions &Opts)
+      : LinuxTargetInfo<SPIR32SYCLDeviceTargetInfo>(Triple, Opts) {
+    LongDoubleWidth = LongDoubleAlign = 128;
```

Comment:
This doesn't seem right, is it?  Linux 32 bit is going to have its long double width modified a bunch, isn't it?  Theres a bunch of flags to change this size.